java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java

```diff
@@ -1205,4 +1204,10 @@ private JavaType getRSubstitution(Result result, String symbolName) {
     return typeSubstitution.substitutedTypes().get(1);
   }
 
+  @Test
+  public void infer_from_type_variable_bounds() throws Exception {
+    Result result = Result.createFor("TypeVarBoundsInference");
+    assertThat(result.symbol("write1").usages()).hasSize(1);
+    assertThat(result.symbol("write2").usages()).hasSize(1);
```

Comment:
Add a test on the parameterized method using enums from file `TypeVarBoundsInference.java`, as you used it in your resolution.